polars/polars-core/src/chunked_array/ops/interpolate.rs

```diff
@@ -71,12 +71,22 @@ where
                                     // another null
                                     Some(None) => {}
                                     Some(Some(high)) => {
-                                        let diff = high - low;
                                         let steps_n = T::Native::from_u32(steps).unwrap();
-                                        for step_i in 1..steps {
-                                            let step_i = T::Native::from_u32(step_i).unwrap();
-                                            let v = linear_itp(low, step_i, diff, steps_n);
-                                            av.push(v)
+                                        if high >= low {
```

Comment:
At the start of this function we could write a match case where we check the `signed` of the variables and assign a different function that does this logic;

```rust
match dtype {
   UInt32 | UInt64 => func_a,
   _ => func_b
}

```

Where the `func_a` and `func_b` have the same arguments and types. 


If you want, I can make a PR on your branch?

Reply:
This would require duplicating the entire function body, right? Is there way to incorporate the type check into the existing conditional such that it can be eliminated in signed cases at compile time? Or perhaps in the existing match?

You would be welcome to make a PR to the branch, or I could do revisions later today.

Reply:
No, not at all. It would have no duplication and no branching in the hot loop. My evening is almost over, so I will take a look tomorrow. It is not the easiest thing if you just started with Rust. :)